source/common/grpc/async_client_impl.h

```diff
@@ -49,9 +49,10 @@ class AsyncClientImpl final : public AsyncClient<RequestType, ResponseType> {
 
   AsyncStream<RequestType>* start(const Protobuf::MethodDescriptor& service_method,
                                   AsyncStreamCallbacks<ResponseType>& callbacks) override {
+    const Optional<std::chrono::milliseconds> no_timeout;
```

Comment:
The `Grpc::AsyncStreamImpl` was not owning the timeout in period between object construction and passing to the `Http::AsyncClient::Stream` in `initialize()`, which is what triggered the ASAN fail.